### PR TITLE
stake: Remove redelegate instruction

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The stake redelegate instruction will not enabled in its current state.

#### Summary of Changes

Since it's been part of the SDK and used downstream, it's a bit finnicky to just deprecate and remove, so here's what was done:

* sdk: deprecate the instruction and stake flag without deprecating the stake flags as a whole
* cli: remove the new command
* program: remove the instruction processor, and all logic that uses the feature gate and stake flag
* transaction-status: keep the instruction parser

Fixes #

### CODE REVIEW

File: sdk/program/src/stake/instruction.rs
Comment (CriesofCarrots) [inline sdk/program/src/stake/instruction.rs]: Is this needed, given the module-level allow at top?

File: cli/src/stake.rs
Comment (CriesofCarrots) [inline cli/src/stake.rs]: Given how careful we're being about deprecating the stake-program sdk stuff, I wonder if we should deprecate this subcommand -- ie hide it and print an error on usage -- rather than removing it wholesale.

**Review by CriesofCarrots — COMMENTED** [review]
Should we remove the feature definition?

Generally, this looks good to me. Two small questions.

File: sdk/program/src/stake/instruction.rs
Comment (joncinque) [reply sdk/program/src/stake/instruction.rs]: Ah good point. There's a few of those in the file, so I'll remove them all

File: cli/src/stake.rs
Comment (joncinque) [reply cli/src/stake.rs]: Yeah sure, no problem! I was thinking that it was fine in this case since it was never released, but it's nicer to get an informative error message than `error: Found argument '...' that wasn't expected`

**Comment by joncinque** [comment]
> Should we remove the feature definition?

I thought we kept around feature flags for longer to check when things were enabled, but since it was never enabled, you're right, it's probably better to just remove it.

File: cli/src/stake.rs
Comment (CriesofCarrots) [inline cli/src/stake.rs]:
```suggestion
                    Arg::with_name("arg").multiple(true).hidden(hidden_unless_forced()),
```
(CI complaint)

**Review by CriesofCarrots — COMMENTED** [review]
lgtm once ci is happy!

File: cli/src/stake.rs
Comment (joncinque) [reply cli/src/stake.rs]: TIL that we needed to do this, thanks!

**Comment by 2501babe** [comment]
super excited for this, will review tomorrow

**Review by 2501babe — APPROVED** [review]

**Review by CriesofCarrots — APPROVED** [review]

**Comment by joncinque** [comment]
@CriesofCarrots, in case you wanted to approve, I just rebased after landing https://github.com/solana-labs/solana-program-library/pull/7033, no other changes

**Comment by joncinque** [comment]
Dang, you beat me to it, thanks!

**Pull request merged**
